### PR TITLE
Add fixed country code feature

### PR DESCRIPTION
**Pull request by keithtxw**
**Changes**
[IMPORTANT] Merge PR[#91](https://github.com/LifeSG/react-design-system/pull/91) before reviewing this

Addition of fixed country code feature for `PhoneNumberInput`

- [delete] branch

**Changelog entry**

- Add fixed country code feature for `PhoneNumberInput`

**Additional information**

- You may refer to this [ticket](https://jira.ship.gov.sg/browse/MOL-12260)

### CODE REVIEW

File: src/phone-number-input/phone-number-input.tsx [hunk 1]
```diff
@@ -118,6 +120,40 @@ export const PhoneNumberInput = ({
         };
     };
 
+    const getAddonProps = (): AddonProps<CountryValue, string> => {
+        if (fixedCountry) {
+            return {
+                type: "label",
+                attributes: {
+                    value: selectedCountryCode,
+                } as LabelAddon,
```
Comment (qroll) [inline src/phone-number-input/phone-number-input.tsx:129]: no need to cast here?

File: src/phone-number-input/phone-number-input.tsx [hunk 2]
```diff
@@ -118,6 +120,40 @@ export const PhoneNumberInput = ({
...
+                    searchPlaceholder: optionSearchPlaceholder,
+                    valueExtractor: (option) => `+${option.countryCode}`,
+                    listExtractor: (option) => ({
+                        title: option.name,
+                        secondaryLabel: `+${option.countryCode}`,
+                    }),
+                    onSelectOption: handleSelectOption,
+                    onHideOptions: onHideOptions,
+                    onShowOptions: onShowOptions,
+                } as ListAddon<CountryValue, string>,
```
Comment (qroll) [inline src/phone-number-input/phone-number-input.tsx:149]: no need to cast here?

File: src/phone-number-input/phone-number-input-helper.tsx [hunk 3]
```diff
@@ -81,7 +81,7 @@ export namespace PhoneNumberInputHelper {
             },
             {
                 formattedText: "",
-                remainingText: text.split(""),
+                remainingText: text ? text.split("") : "",
```
Comment (qroll) [inline src/phone-number-input/phone-number-input-helper.tsx:84]: the logic still works with string, but for correctness, it should be initialised to an empty array
```suggestion
                remainingText: text ? text.split("") : [],
```

Comment (qroll) [inline src/phone-number-input/phone-number-input-helper.tsx:84]: can add a unit test for calling `formatNumber` with an empty number?

**Review by qroll — CHANGES_REQUESTED** [review]

File: src/phone-number-input/phone-number-input.tsx (hunk 1 above)
Comment (keithtxw) [reply src/phone-number-input/phone-number-input.tsx:129]: We need to else I will be able to pass list addon props

File: src/phone-number-input/phone-number-input.tsx (hunk 2 above)
Comment (keithtxw) [reply src/phone-number-input/phone-number-input.tsx:149]: same as above. vice versa
